### PR TITLE
Add e2e test

### PR DESCRIPTION
**Pull request by suttang**
## Motivation and Context

In blueoil rebuilding project, we planned to replace e2e test script from sh to python.

## Description

- Move some operations in `blueoil/cmd/main.py` to each scripts to improve testability.
- Create `tests/e2e`

~~You don't need to check `*.yml` files, it is the copy of the current test yaml files.~~

## How has this been tested?

```
DATA_DIR=/storage/dataset CUDA_VISIBLE_DEVICES=4 docker-compose run --rm blueoil pytest tests/e2e/
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by iizukak** [comment]
@suttang Why did you create new yaml file instead of use `git mv` ?

File: blueoil/cmd/main.py [hunk 1]
```diff
@@ -172,14 +153,7 @@ def convert(experiment_id, checkpoint, template, image_size, project_name):
...
 def predict(input, output, experiment_id, checkpoint, save_images):
-    output_dir = os.environ.get('OUTPUT_DIR', 'saved')
-
-    if checkpoint is None:
-        restore_path = None
-    else:
-        restore_path = os.path.join(output_dir, experiment_id, 'checkpoints', checkpoint)
-
-    run_predict(input, output, experiment_id, None, restore_path, save_images)
+    run_predict(input, output, experiment_id, checkpoint, save_images)
```
Comment (iizukak) [inline blueoil/cmd/main.py]: `preduct` function looks pass through argument to `run_predict`. 
The worth of existence of this function is `click.echo` ?
Or You wan to unify interface with `convert` ?

**Comment by suttang** [comment]
@iizukak OH!!! I did not know the test yaml files already existed.
I will fix them!

File: blueoil/cmd/main.py (hunk 1 above)
Comment (suttang) [reply blueoil/cmd/main.py]: Both are correct.
To improve testability, I separated the logic and interface.
The main file is responsible for input from the user and sends message to the user.

File: blueoil/cmd/main.py [hunk 2]
```diff
@@ -73,25 +71,16 @@ def init(output):
...
-        experiment_id = 'train_{:%Y%m%d%H%M%S}'.format(datetime.now())
-
-    run_train(config, experiment_id)
-
-    output_dir = os.environ.get('OUTPUT_DIR', 'saved')
-    experiment_dir = os.path.join(output_dir, experiment_id)
-    checkpoint = os.path.join(experiment_dir, 'checkpoints', 'checkpoint')
-    if not os.path.isfile(checkpoint):
-        click.echo('Checkpoints are not created in {}'.format(experiment_dir), err=True)
+    try:
```
Comment (hadusam) [inline blueoil/cmd/main.py]: Why we need a `try-catch` for only `train` part?
How about `convert` and `predict`?

Comment (suttang) [reply blueoil/cmd/main.py]: I moved [this error handling operation](https://github.com/blue-oil/blueoil/pull/536/files#diff-1ce1d3673b3c8bd027df36f98c119f8aL84-L85) to [here](https://github.com/blue-oil/blueoil/pull/536/files#diff-0e36593a8447a90b2de26bc526b2255bR75-R76).

Yes, we should use `try-catch` for `convert` and `predict`.
but in this PR, I did not refactor `convert` and `predict`.

Comment (hadusam) [reply blueoil/cmd/main.py]: Okay, I see. 👍

**Comment by iizukak** [comment]
@tkng Need readability reviewer.

**Comment by suttang** [comment]
@tkng please review when you have time.

**Comment by tkng** [comment]
- [x] readability approval

**Pull request merged**